### PR TITLE
Expose isPressed on useLink hook

### LINKED ISSUE #1766: Expose isPressed in useLink return
# 🙋 Feature Request

When I'm using `useLink` I expect the same behavior of `useButton` related to the `isPressed` state.

## 🤔 Expected Behavior

Expose `isPressed` state.

## 😯 Current Behavior

Do not expose `isPressed` state.

## 💁 Possible Solution

Expose `isPressed` state.

## 🔦 Context

I'm building a Link component that receives `as` prop (to replace `a` tag with another element) and I need to keep the same behavior as `a` has with css `:active` state.

### PR DESCRIPTION
**Pull request by hyanmandian**
Closes #1766 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by LFDanLu — COMMENTED** [review]
IMO this seems reasonable to me and maybe something we'll want to do for our other hooks that use `usePress` as well. Lemme bring this up with the rest of the team in case they had any other thoughts

File: packages/@react-aria/link/docs/useLink.mdx
Comment (LFDanLu) [inline packages/@react-aria/link/docs/useLink.mdx:112]: I noticed this `darkblue` is rather hard to see in dark mode. What do you think about using something like `var(--spectrum-global-color-orange-400)` or one of the other colors [here](https://react-spectrum.adobe.com/react-spectrum/styling.html) as the press/non-pressed color? 

Note you'll need to inspect the DOM to see the global variable names actually being used on that styling page

**Review by LFDanLu — CHANGES_REQUESTED** [review]

File: packages/@react-aria/link/docs/useLink.mdx
Comment (hyanmandian) [reply packages/@react-aria/link/docs/useLink.mdx:112]: Nice idea @LFDanLu, I will do this change.

Comment (hyanmandian) [reply packages/@react-aria/link/docs/useLink.mdx]: Done! I just changed to `var(--spectrum-global-color-blue-700)` instead of  `var(--spectrum-global-color-orange-400)` because have better contrast ratio in light mode.

Thanks!

Comment (hyanmandian) [reply packages/@react-aria/link/docs/useLink.mdx]: Done! I just changed to `var(--spectrum-global-color-blue-700)` instead of  `var(--spectrum-global-color-orange-400)` because have better contrast ratio in light mode.

Thanks!

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks!

**Review by snowystinger — APPROVED** [review]
This looks good to me, will need to wait for API sign off though before merging.

**Review by devongovett — APPROVED** [review]
Ok. This seems like a fairly common use case so I think it's ok. But I don't want this to lead to a slippery slope of adding states to every hook. It's fairly straightforward to use the `onPressChange` event to do this yourself, for example, so on other components I think I'd suggest that rather than adding to the API for every component.

**Pull request merged**
